cogs/banner.py

```diff
@@ -0,0 +1,548 @@
+# Copyright (C) idoneam (2016-2021)
+#
+# This file is part of Canary
+#
+# Canary is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Canary is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Canary. If not, see <https://www.gnu.org/licenses/>.
+
+# discord-py requirements
+import discord
+from discord.ext import commands, tasks
+from discord import utils
+
+# Other utilities
+from io import BytesIO
+import datetime
+import sqlite3
+import requests
+from PIL import Image, UnidentifiedImageError
+import json
+from .utils.checks import is_moderator
+import asyncio
+
+
+class Banner(commands.Cog):
+    # Written by @le-potate
+    def __init__(self, bot):
+        self.bot = bot
+        self.guild = None
+        self.banner_role = None
+        self.banner_channel = None
+        self.preview_channel = None
+        self.converted_channel = None
+        self.bots_channel = None
+        self.winner_role = None
+        self.redchiken = None
+        self.start_datetime = None
+        self.week_name = None
+        self.send_reminder = None
+
+    @commands.Cog.listener()
+    async def on_ready(self):
+        self.guild = self.bot.get_guild(self.bot.config.server_id)
+        self.banner_channel = utils.get(self.guild.text_channels,
```

Comment:
i think all of these elements should be set as options in `config.ini`, it would also mean you don't need to set these options to `None` in the constructor

Reply:
Resolved in d48fa5f